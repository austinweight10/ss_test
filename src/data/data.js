export default [
  {
    title: "Nike Air Max 98",
    image: "https://cdn.thesolesupplier.co.uk/2017/05/AIR-MAX-98-2.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/air-max-98/",
  },
  {
    title: "Nike Air VaporMax Plus",
    image: "https://cdn.thesolesupplier.co.uk/2017/05/VaporMax-Plus-.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/air-vapormax-plus/",
  },
  {
    title: "Skepta x Nike Sk Air",
    image: "https://cdn.thesolesupplier.co.uk/2017/05/Skepta-Sk-Air-nike.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/skepta-nike-sk-air/",
  },
  {
    title: "Converse One Star",
    image:
      "https://cdn.thesolesupplier.co.uk/2018/08/Carhartt-x-Converse-One-Star-WIP-White-09.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/converse/one-star/",
  },
  {
    title: "Yeezy 700",
    image: "https://cdn.thesolesupplier.co.uk/2019/09/Inertia2.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/yeezy/700/",
  },
  {
    title: "Nike Air Max 270 React",
    image:
      "https://cdn.thesolesupplier.co.uk/2019/06/Nike-Air-Max-270-React-AO4971-002-5.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/air-max-270-react/",
  },
  {
    title: "Nike Air Jordan 1",
    image:
      "https://cdn.thesolesupplier.co.uk/2019/06/Jordan-1-Shattered-Backboard-3.0-555088-028-lifestyle.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/air-jordan-1/",
  },
  {
    title: "Nike Air Force 1",
    image: "https://cdn.thesolesupplier.co.uk/2019/02/AIR-FORCE-1-4.jpg",
    url: "https://thesolesupplier.co.uk/release-dates/nike/air-force-1/",
  },
]
