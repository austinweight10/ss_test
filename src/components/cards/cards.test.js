import React from "react"
import Enzyme, { shallow } from "enzyme"

import Adapter from "enzyme-adapter-react-16"

import Cards from "./"
import data from "../../data/data"

Enzyme.configure({ adapter: new Adapter() })

describe("Cards", () => {
  const wrapper = shallow(<Cards data={data} />)

  it("Card component contains 8 cards", () => {
    expect(wrapper.find("[data-test='card_container']").length).toEqual(8)
  })

  it("Card component 1 link", () => {
    expect(
      wrapper
        .find("[data-test='card_container']")
        .at(0)
        .prop("href")
    ).toEqual(data[0].url)
  })

  it("Card component 1 image", () => {
    expect(
      wrapper
        .find("[data-test='card_image']")
        .at(0)
        .prop("image")
    ).toEqual(data[0].image)
  })

  it("Card component 1 title", () => {
    expect(
      wrapper
        .find("[data-test='card_title']")
        .at(0)
        .text()
    ).toEqual(data[0].title)
  })

  it("Card matches shapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
