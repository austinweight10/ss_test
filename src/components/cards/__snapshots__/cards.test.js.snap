// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Card matches shapshot 1`] = `ShallowWrapper {}`;
